universe/Universe.cpp

```diff
@@ -1659,7 +1659,11 @@ void Universe::ApplyEffectDerivedVisibilities() {
         for (const auto& object_entry : empire_entry.second) {
             if (object_entry.first <= INVALID_OBJECT_ID)
                 continue;
-            m_empire_object_visibility[empire_entry.first][object_entry.first] = object_entry.second;
+            // TODO: If/when the parser for SetVisibility is expanded to allow ValueRefs like
+            // max(Target.visibility, Basic)
+            // then the folllowing should be reverted to a simple assignment.
+            m_empire_object_visibility[empire_entry.first][object_entry.first] = 
+                std::max(object_entry.second, m_empire_object_visibility[empire_entry.first][object_entry.first]);
```

Comment:
As mentioned in #1708, I think that `SetMinVisibility()` and `SetMaxVisibility()` are clearer and more functional, (although not as functional as something evaluated like a ValueReef).  With this change setting visibility from an effect can only act as a set minimum visibility.

I don't object to merging this PR, which is a step forward and fixes the bug.